### PR TITLE
Update design system to latest version

### LINKED ISSUE #1981: Implement Updated CMS Design System
We should be able to implement the CMS Design System as it gets updated without losing functionality within the application.

### This task is done when...
 - [ ] New CMS Design System has been implemented
 - [ ] We are able to have a path/process for updating the Design System as it gets released (and it is documented or automated - like NASA)

### PR DESCRIPTION
**Pull request by mgwalker**
### This pull request changes...

- updates the design system to 3.4.2
- replaces `<Select>` with `<Dropdown>`
- switches `fieldRef` to `inputRef` and swaps it with a function, since that's now required by the design system libraries
- updates `autoFocus` props to be booleans instead of strings, to stop React complaining
- updates the link colors in the header to include a `:visited` pseudoselector so they don't turn purple and disappear
- closes #1981

### This pull request is ready to merge when...

- [x] Tests have been updated (and all tests are passing)
- [x] This code has been reviewed by someone other than the original author
- [x] The experience passes a basic manual accessibility audit (keyboard nav, screenreader, text scaling) OR an exemption is documented
- ~The change has been documented~ N/A
  - ~Associated OpenAPI documentation has been updated~ N/A
- [x] Changelog is updated as appropriate

### This feature is done when...

Basically need a quick run through to make sure things still look and behave the same!

- [ ] Design has approved the experience
- [ ] Product has approved the experience

### CODE REVIEW

File: web/src/containers/activity/CostAllocateFFP.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import { ChoiceList } from '@cmsgov/design-system-core';
```
Comment (hbillings) [inline web/src/containers/activity/CostAllocateFFP.js]: I hadn't even realized there were two ways of doing this. Interesting!
